File: scrimage-core/src/test/scala/com/sksamuel/scrimage/X11ColorlistGenerator.scala
package com.sksamuel.scrimage

import org.apache.commons.lang3.text.WordUtils
import scala.io.Source


object X11ColorlistGenerator extends App {
  val file = "/etc/X11/rgb.txt"
  Using.resource(Source.fromFile(file)) { lines =>
    for (line <- lines.getLines()) {
      val tokens = line.replaceAll("\\s{2,}", " ").trim.split(' ')
      val red    = tokens(0)
      val green  = tokens(1)
      val blue   = tokens(2)
      val name   = WordUtils.capitalize(tokens.drop(3).mkString(" "))
      if (!name.contains(" "))
        println(s"val $name = Color($red, $green, $blue)")
    }
  }
}

File: scrimage-tests/src/test/kotlin/com/sksamuel/scrimage/core/X11ColorlistGenerator.kt
package com.sksamuel.scrimage.core

import java.nio.file.Files
import java.nio.file.Paths

fun main() {
   val file = "/etc/X11/rgb.txt"
   Files.newInputStream(Paths.get(file)).bufferedReader().readLines().map { line ->
      val tokens = line.replace("\\s{2,}".toRegex(), " ").trim().split(' ')
      val red = tokens[0]
      val green = tokens[1]
      val blue = tokens[2]
      val name = tokens.drop(3).joinToString(" ").trim()
      if (!name.contains(" "))
         println("val $name = Color($red, $green, $blue)")
   }
}
